initial impl of stubbed service

daiLlew opened the pull request:
MVP implementation of stub service

daiLlew committed:
initial impl of stubbed service

daiLlew committed:
minor refactor and started updating the README

daiLlew committed:
Update README.md

daiLlew committed:
now acts on both service and user headers

daiLlew committed:
added more scenarios

Matt-Guest commented on identity/handler.go:
```diff
@@ -0,0 +1,76 @@
+		return nil, err
+	}
+
+	b, _ := json.MarshalIndent(profiles, "", " ")
```
The error type should be checked here.

Matt-Guest commented on identity/handler.go:
```diff
@@ -0,0 +1,76 @@
+	}
+
+	b, _ := json.MarshalIndent(profiles, "", " ")
+	log.Debug(string(b), nil)
```
Debug message here

Matt-Guest commented on main.go:
```diff
@@ -0,0 +1,60 @@
+)
+
+func main() {
+	log.HumanReadable = true
```
Should this be set from the Makefile?

daiLlew replied:
I dont know. I didn't bother with an of the CI stuff because its stub - I assume this is for local testing so you can just run it without all that gumpf

gedge committed:
add config/config.go to allow bind address to be configured

Matt-Guest reviewed (commented):
Can a Makefile be added

LloydGriffiths commented on config.yml:
```diff
@@ -0,0 +1,34 @@
+  - scenario: "authorized service: dataset-api, authorized user"
+    authorization-token: "939616dc-7599-4ded-9a86-a9c66fbf98e0"
+    x-florence-token: "85c718c3-9ba4-4f31-99bb-3e4eaabb2cc1"
+    identifier: "rickSanchez@theCitadelOfRicks.com"
```
🎉

CarlHembrough commented:
Dependencies need vendoring

gedge committed:
move shutdown to end

daiLlew committed:
added support for create service account

daiLlew committed:
fixed merge conflicts

daiLlew committed:
Update README.md

nshumoogum committed:
New auth tokens to match service defaults

This will allow end to end tests to pass as import services will be able
to communicate to private endpoints without receiving a 401.

Export services yet to be added

nshumoogum committed:
Add Makefile

nshumoogum committed:
Update README.md

Also remove binary file

nshumoogum committed:
Add vendoring to repo

nshumoogum committed:
Update test scenarios and extend

scenarios are now clearly described

Matt-Guest commented on handler/api_stub.go:
```diff
@@ -0,0 +1,116 @@
+
+	b, err := json.MarshalIndent(scenarios, "", " ")
+	if err != nil {
+		return nil, errors.New("failed to marshal config to json")
```
Can the error be logged here

nshumoogum replied:
Yes it can......but am i going to do it!.........................okay i've decided to do it

nshumoogum committed:
Logout error before returning from method

Also add comments to exported functions and types
